sui/token_bridge/sources/create_wrapped.move

```diff
@@ -118,4 +121,38 @@ module token_bridge::create_wrapped {
             decimals,
         );
     }
+
+    public entry fun update_registered_wrapped_coin_metadata<CoinType>(
```

Comment:
currently the `CoinType` is not verified which means that the coin that the VAA refers to might be different from `CoinType`. this allows an attacker to grab an attestation VAA and run it against any wrapped coin type. we should query `state::token_info<CoinType>` and ensure that the token chain and token address match what's in the VAA

Reply:
that check should probably exist in `state::update_registered_wrapped_coin_metadata<CoinType>`